code/game/dna/mutations/mutation_powers.dm

```diff
@@ -209,19 +209,26 @@
 	..()
 	block = GLOB.shadowblock
 
+/datum/mutation/stealth/darkcloak/deactivate(mob/living/M)
+	..()
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.set_alpha_tracking(ALPHA_VISIBLE, src)
+
 /datum/mutation/stealth/darkcloak/on_life(mob/M)
 	var/turf/simulated/T = get_turf(M)
-	if(!istype(T))
+	if(!istype(T) || !ishuman(M))
 		return
+	var/mob/living/carbon/human/H = M
 	var/light_available = T.get_lumcount() * 10
 	if(light_available <= 2)
-		if(M.invisibility != INVISIBILITY_LEVEL_TWO)
-			M.alpha = round(M.alpha * 0.8)
+		if(H.invisibility != INVISIBILITY_LEVEL_TWO)
+			H.set_alpha_tracking(H.get_alpha() * 0.8, src)
 	else
-		M.reset_visibility()
-		M.alpha = round(255 * 0.8)
-	if(M.alpha == 0)
-		M.make_invisible()
+		H.reset_visibility()
+		H.set_alpha_tracking(ALPHA_VISIBLE * 0.8, src)
+	if(H.get_alpha(src) == 0)
+		H.make_invisible()
```

Comment:
make_invisible directly sets the mob's alpha to 128, will that cause problems for this system?

Reply:
No, that alpha will be reset when they become non-invisible